src/pages/Signup.tsx

```diff
@@ -1,6 +1,163 @@
 import React from 'react';
-import { Text } from 'react-native';
+import { Pressable, StyleSheet, Text, TextInput, View } from 'react-native';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import { NavigationProp } from '@react-navigation/core';
+import { CustomButton } from '../components/CustomButton';
+import { signupRequest } from '../services/user.services';
+import { useToastAlert } from '../hooks/useToastAlert';
 
-export const Signup = () => {
-  return <Text>Signup</Text>;
+interface SignupProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  navigation: NavigationProp<any, any>;
+}
+
+export const Signup = ({ navigation }: SignupProps) => {
+  const { showInfoToast, showErrorToast } = useToastAlert();
+
+  const redirectToLogin = () => {
+    navigation.navigate('Login');
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      email: '',
+      username: '',
+      password: ''
+    },
+    validationSchema: Yup.object({
+      email: Yup.string().email().required(),
+      username: Yup.string().required(),
+      password: Yup.string()
+        .matches(
+          /^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[$!%*#?&/%])[A-Za-z\d$!%*#?&/%]{8,}$/,
+          'Password must be at least 8 characters long, contain at least one uppercase letter, one lowercase letter, one number, and one special character'
+        )
+        .required()
+    }),
+
+    onSubmit: async (values) => {
+      const [response, error] = await signupRequest(
+        values.email,
+        values.username,
+        values.password
+      );
+      if (error && response?.message) {
+        showErrorToast(response?.message);
+      } else {
+        showInfoToast('User created succesfully, now confirm your Email');
```

Comment:
@SilviaPabon I think it's better to show a success alert instead of an information one.